dashboard/app/models/learning_module.rb

```diff
@@ -0,0 +1,40 @@
+require 'securerandom'
+# == Schema Information
+#
+# Table name: learning_modules
+#
+#  id                              :integer          not null, primary key
+#  name                            :string(255)
+#  learning_module_type            :string(255)
+#  created_at                      :datetime         not null
+#  updated_at                      :datetime         not null
+#  professional_learning_course_id :integer
+#
+# Indexes
+#
+#  index_learning_modules_on_professional_learning_course_id  (professional_learning_course_id)
+#
+
+class LearningModule < ActiveRecord::Base
+  has_many :artifacts, class_name: 'Artifact', dependent: :destroy
+  belongs_to :professional_learning_course
+
+  def self.create_from_params(name, learning_module_type, course)
```

Comment:
create != find_or_create_by so the custom method here is actively surprising in it's behavior.  

If you want to ensure that all required params are provided, I think it would be better to make this find_or_create_by_params (using named  parameters).   

I also think seeing the following in client code would be perfectly in line with what we do elsewhere;  Rails just isn't a strongly typed language so there's not too much point in trying to fight that.
```
LearningModule.find_or_create_by(name: name, learning_module_type: learning_module_type, professional_learning_course: course)
```